lib/autoupdate/tools/updater.go

```diff
@@ -421,8 +423,13 @@ func (u *Updater) Exec(ctx context.Context, toolsVersion string, args []string)
 		teleportToolsVersionReExecEnv,
 		teleportToolsDirsEnv,
 	})
-	env = append(env, fmt.Sprintf("%s=%s", teleportToolsVersionReExecEnv, u.localVersion))
-	env = append(env, fmt.Sprintf("%s=%s", teleportToolsDirsEnv, u.toolsDir))
+	env = append(env, teleportToolsVersionReExecEnv+"="+u.localVersion)
+	env = append(env, teleportToolsDirsEnv+"="+u.toolsDir)
+	// If tsh or tctl has already been re-executed with the original path,
+	// we need to pass that path to the next re-execution.
+	if GetReExecPath() == "" {
+		env = append(env, teleportToolsPathReExecEnv+"="+executablePath)
+	}
```

Comment:
This is relying on the `teleportToolsPathReExecEnv` envvar being inherited, right? Can we filter `teleportToolsPathReExecEnv` from environ and always set it explicitly instead?